File: 7-kyu/do-you-speak-retsec-/python/solution.py
def reverse_by_center(s):
    midpt = len(s) // 2 if len(s) % 2 == 1 else 0
    return s[midpt + 1:] + s[midpt] + s[:midpt] if midpt else s[(len(s) + 1) // 2:] + s[:(len(s) + 1 )// 2]

File: 7-kyu/do-you-speak-retsec-/javascript/solution.js
const reverseByCenter = s => {
  const firstHalf = s.slice(0, Math.floor(s.length / 2));
  const secondHalf = s.slice(Math.ceil(s.length / 2));
  const middle = s.length % 2 === 1 ? s[Math.ceil(s.length / 2) - 1] : "";
  return `${secondHalf}${middle}${firstHalf}`;
};
